src/containers/guest-home-page/google-button/GoogleButton.jsx

```diff
@@ -12,29 +13,31 @@ import { snackbarVariants } from '~/constants'
 import { styles } from '~/containers/guest-home-page/google-button/GoogleButton.styles'
 
 const GoogleButton = ({ role, route, buttonWidth, type }) => {
+  const ref = useHref(route)
   const dispatch = useDispatch()
   const mediaQuery = useBreakpoints().isLaptopAndAbove ? 'md' : 'xs'
   const { closeModal } = useModalContext()
   const { setAlert } = useSnackBarContext()
-  const ref = useHref(route)
+  const [googleAuth] = useGoogleAuthMutation()
 
   const handleCredentialResponse = useCallback(
     async (token) => {
       try {
-        await dispatch(googleAuth({ token, role })).unwrap()
+        const response = await googleAuth({ token, role }).unwrap()
+        dispatch(setUser(response.accessToken))
```

Comment:
I suggest doing all in the slice.. so we only call googleAuth and all is doing on the background...

Reply:
I agree with you, used onQueryStarted for that, and doing dispatch when query is fullfilled